lib/new_relic/agent/transaction/trace_context.rb

```diff
@@ -148,6 +146,29 @@ def accept_trace_context(header_data)
           false
         end
 
+        def determine_sampling_decision(payload, header_data)
+          unless payload.sampled.nil?
+            if header_data.trace_parent['trace_flags'] == '01'
```

Comment:
I believe [trace-flags is a bitmask](https://www.w3.org/TR/trace-context/#trace-flags), even though `01` is the only value that's currently valid. Do we want to future-proof it now, or not worry about it yet?

Reply:
Yeah, I think that's a good question. The reason I decided to check if it's 01 and then just else was because if it's anything else (like some weird value or even some future unknown value), I figured falling back to the not sampled logic makes the most sense. Since any new value would require us to implement whatever logic is required for the new behavior tied to that value, we'll need to make changes anyways though. 

I'm open to swapping the logic and defaulting to assuming everything is sampled if it's not 00 if we think that would be better, I was just thinking that assuming not sampled would work out better, for max samples stored reasons, in most situations if we don't know what it is.

Reply:
Yeah, i think based on the latest thing from chris about the customer issue, with the "If there is no valid traceparent, then the legacy priority sampling logic will be maintained." thing, keeping it the way it is currently is the right call.

Reply:
hmm or actually. i think i might make a change now that i'm thinking about it again

Reply:
Is there a scenario where we'd receive a value in `header_data.trace_parent['trace_flags']` that wouldn't `==` to `01`, but should be marked as sampled? Would we ever receive `0x01` or something else?